app/routes/users.js

```diff
@@ -46,39 +46,53 @@ function Route(app) {
     .post(jwt({secret: TOKEN_SECRET}), auth.ensureAdmin, userController.create.bind(userController));
 
   // Route for operations that target individual users
-  router.route('/api/v0/users/:User.:format?')
+  userRouter.route('/:User.:format?')
     .get(jwt({secret: TOKEN_SECRET}), auth.ensureAdmin, userController.get.bind(userController))
     .put(jwt({secret: TOKEN_SECRET}), auth.ensureAdmin, userController.update.bind(userController))
     .delete(jwt({secret: TOKEN_SECRET}), auth.ensureAdmin, userController.remove.bind(userController));
 
-  app.use(router);
+
+  // Create User settings routes
+  const settingsRoute = express.Route();
+  settingsRoute.route('/:Namespace')
+    .get(jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, userController.getSettings.bind(userController))
+    .put(jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, userController.updateSettings.bind(userController))
+    .delete(jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, userController.deleteSettings.bind(userController));
+  userRouter.use('/settings', settingsRoute);
+
+  // allows to use /client-settings instead of /settings
+  userRouter.use('/client-settings/', express.Route().all((req) => { req.redirect('../settings'); }));
 
   // Create authentication routes:
   app.get('/api/v0/apikeys', jwt({secret: TOKEN_SECRET}), auth.ensureAdmin, apiKeys.keys);
-  app.get('/api/v0/users/:User/apikeys', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, apiKeys.userKeys);
-  app.get('/api/v0/users/:User/apikeys/new', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, apiKeys.createKey);
-  app.delete('/api/v0/users/:User/apikeys/:Key', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated,
-    apiKeys.deleteKey);
-
-  app.post('/auth/login', authController.login.bind(authController));
-  app.get('/auth/me', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated,
-    authController.getme.bind(authController));
-  app.put('/auth/me', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated,
-    authController.putme.bind(authController));
-  app.post('/auth/signup', authController.signup.bind(authController));
-  app.post('/auth/logout', authController.logout.bind(authController));
-  app.post('/auth/github', jwt({secret: TOKEN_SECRET, credentialsRequired: false}), AuthController.github);
-  app.post('/auth/google', authController.google.bind(authController));
-  app.get('/auth/github/id', AuthController.getGithubClientId);
+  userRouter
+    .get('/apikeys', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, apiKeys.userKeys)
+    .get('/apikeys/new', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, apiKeys.createKey)
+    .delete('/apikeys/:Key', jwt({secret: TOKEN_SECRET}), auth.ensureAuthenticated, apiKeys.deleteKey);
+
+  // register user router
+  app.use('/api/v0/users', userRouter);
+
+  const authRoute = express.Route();
+  authRoute
```

Comment:
Could this auth have a separate route file, since logically it already has its own router.